{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "import viprs as vp\n",
    "import magenpy as mgp\n",
    "import numpy as np\n",
    "import zarr\n",
    "from sklearn.model_selection import KFold\n",
    "from sklearn.model_selection import StratifiedGroupKFold\n",
    "import pandas as pd "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.random.seed(42)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 23.03it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 386.39it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 224.13it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 36.03it/s]\n"
     ]
    }
   ],
   "source": [
    "gdl = mgp.GWADataLoader(bed_files=\"Japan_ASD/japan_chr22/*\", \n",
    "                        sumstats_files=\"GWAS/chr22.tsv\",\n",
    "                        sumstats_format=\"GWASCatalog\",\n",
    "                        ld_store_files=\"LD_EAS/chr_22/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# array of sample IDs \n",
    "sample_ids = gdl.samples "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://melaniewalsh.github.io/Intro-Cultural-Analytics/02-Python/EXTRA-Randomize-Students.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "# code for selecting random CV folds \n",
    "def get_folds(samples, n_folds, seed):\n",
    "    np.random.seed(seed)\n",
    "    np.random.shuffle(sample_ids)\n",
    "    all_folds = [] \n",
    "    for i in range(n_folds):\n",
    "        group = samples[i::n_folds]\n",
    "        all_folds.append(group)\n",
    "    return all_folds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "folds = get_folds(sample_ids, 5, 42)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To-do: expand this to all chromosomes "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_cv_prs(chrom):\n",
    "    np.random.seed(42)\n",
    "    df_list = []\n",
    "    get_folds(sample_ids, 5, 42)\n",
    "    for i in range(5):\n",
    "        training_folds = [item for item in sample_ids if item not in folds[i]]\n",
    "        gdl_train = mgp.GWADataLoader(bed_files=f\"Japan_ASD/japan_chr{chrom}/*\", \n",
    "                        sumstats_files=f\"GWAS/chr{chrom}.tsv\",\n",
    "                        sumstats_format=\"GWASCatalog\",\n",
    "                        ld_store_files=f\"LD_EAS/chr_{chrom}/\",\n",
    "                        keep_samples=training_folds)\n",
    "        gdl_test = mgp.GWADataLoader(bed_files=f\"Japan_ASD/japan_chr{chrom}/*\", \n",
    "                        sumstats_files=f\"GWAS/chr{chrom}.tsv\",\n",
    "                        sumstats_format=\"GWASCatalog\",\n",
    "                        ld_store_files=f\"LD_EAS/chr_{chrom}/\",\n",
    "                        keep_samples=folds[i])\n",
    "        v = vp.VIPRS(gdl_train)\n",
    "        v.fit()\n",
    "        results = v.predict(gdl_test)\n",
    "        df = pd.DataFrame({'ID': folds[i], f\"PRS_Chr{chrom}\": results})\n",
    "        df_list.append(df)\n",
    "    df = pd.concat(df_list)\n",
    "    df_indexed = df.reset_index().drop(columns=\"index\")\n",
    "    return df_indexed "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.63it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 360.61it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 83.38it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  9.76it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.06it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 825.16it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 91.67it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 10.01it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "fa9b39d8be94454e9bef1e4b5f61189f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 1 (410 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  6.14it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  4.56it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 774.00it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 94.67it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  9.89it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.96it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 771.44it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 95.68it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  9.39it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "663f17119db545b889c6241a5f6eda22",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 1 (410 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  8.65it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.75it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 856.85it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 100.13it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  9.67it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.01it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 839.70it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 96.22it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  9.77it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0f57064f05b04edab54332d3ee7cbd01",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 1 (410 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  5.62it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  4.32it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 788.70it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 102.63it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  9.45it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.95it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 864.98it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 93.00it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  9.71it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0ea8c406355d49ecbe8e4d477c74c3f9",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 1 (410 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  6.48it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.06it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 938.11it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 96.75it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  9.56it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.03it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 854.41it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 99.93it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  9.66it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "a087e43d3c3b4d8881830b301f631b29",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 1 (410 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 13.80it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.35it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 276.41it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 62.01it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  8.42it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.75it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 817.60it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 95.13it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  8.99it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "5ddaaff84893441c980e9d9da2726c47",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 2 (465 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  4.98it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  4.52it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 774.00it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 88.00it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  9.32it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.74it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 858.26it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 92.52it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  5.48it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "7783c84e954147508658cced709358c0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 2 (465 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  7.13it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  3.94it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 401.41it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 69.15it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  8.39it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.80it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 903.36it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 84.67it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  8.95it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "fb9bf02a0d7e46a581f29bfa51d70991",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 2 (465 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  7.67it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.46it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 601.33it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 71.81it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  8.62it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.60it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 473.02it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 63.17it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  8.12it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "e12c3d175ce345e2a8a3291750a70938",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 2 (465 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  5.78it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  4.67it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 304.13it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 65.93it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  8.76it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.73it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 742.35it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 94.88it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00,  9.11it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "eb7954ec966843d59f1067c3e73ac777",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 2 (465 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  7.35it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.07it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 262.32it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 77.53it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 10.82it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.88it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 698.47it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 92.19it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 11.25it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "b556e2674bed407f9f88dd6e4376c8c1",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 3 (350 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.010236).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 11.97it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.82it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 773.57it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 87.65it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 10.99it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.96it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 853.19it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 90.35it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 11.10it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d2b329aba6704b93bcc596e7050a0e77",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 3 (350 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.002170).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 15.58it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.86it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 821.77it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 98.60it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 11.56it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.87it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 836.19it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 87.11it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 11.53it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "665242f43fd14659a6588c966bc21c0c",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 3 (350 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.008048).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 15.20it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.80it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 836.85it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 96.86it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 11.51it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.88it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 829.90it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 106.08it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 11.35it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "a8b06155540644efb0c9ad03199bcb7d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 3 (350 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.001928).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.79it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.80it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 788.55it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 91.81it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 10.98it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.81it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 840.37it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 90.32it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 11.26it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "5998161fed0b493d8632928bd8d9d10b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 3 (350 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.008323).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 11.63it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.16it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 333.68it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 98.25it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.91it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.23it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 845.28it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 89.74it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.57it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "4c5bd12cba2a4c32b69c6b523694bbc3",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 4 (347 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  8.93it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.61it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 923.45it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 106.49it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.72it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.30it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 873.27it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 94.07it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.26it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "cb49003667174234aa38ba7f71175415",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 4 (347 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  9.31it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.68it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 941.48it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 102.57it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.40it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.26it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 756.41it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 106.17it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.78it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "64e8548c4e444c818ee200ea16ec889b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 4 (347 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.52it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.52it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 843.92it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 109.90it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.54it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.39it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 747.91it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 106.03it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.48it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f90787073f9143758a699eb7b1f3640d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 4 (347 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  6.79it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.21it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 835.35it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 96.76it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.66it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.21it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 909.04it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 107.93it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.49it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "138375b036e7448ca9a1b20666e16498",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 4 (347 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  8.65it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.47it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 402.25it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 101.08it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.04it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.16it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 729.95it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 110.75it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.31it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "ca1accc80e094108a3b87269766a74e9",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 5 (327 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  8.27it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.44it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 909.04it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 103.62it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.44it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.14it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 911.61it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 108.38it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.54it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "bac0e696cb234e758246cbf56ae822ba",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 5 (327 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  7.88it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.18it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 897.56it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 100.95it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.95it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.16it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 881.90it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 110.99it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.11it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f321a81e775749a6bbd939b5c69fe28c",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 5 (327 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  8.59it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.88it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 927.53it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 102.47it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.18it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.30it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 748.72it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 97.47it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.86it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "8ed9ee72e1834e28a31a2116c9fc78ea",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 5 (327 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  7.81it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.70it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 950.66it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 102.50it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.73it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.33it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 794.23it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 97.51it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 11.69it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "fdaac946c4ac4e6a87703ee6cac3daeb",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 5 (327 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  9.13it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.79it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 405.36it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 108.55it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 11.66it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.31it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 951.52it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 103.19it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 11.85it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d957495584474eed9d60f6ee804bb14b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 6 (310 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  9.84it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  4.30it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 876.55it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 110.39it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.51it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.24it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 839.03it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 111.77it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.21it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "b285f64c147547de8703432277c90564",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 6 (310 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  8.23it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.63it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 336.68it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 86.54it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 11.84it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.34it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 625.36it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 117.25it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.29it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "ddccb7a7d28f4b61ab10c7393b997eaa",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 6 (310 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  9.91it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.93it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 896.22it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 111.83it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.25it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.26it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 855.28it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 103.53it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.31it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "6b10b93de6c14c83919a8df01ef394d5",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 6 (310 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  8.96it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.76it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 548.78it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 91.79it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.11it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.28it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 872.54it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 102.67it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.49it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "9264e3e62cc0411086f5cb7f3adab534",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 6 (310 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  6.65it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.03it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 357.02it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 102.38it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.52it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.60it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 986.20it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 123.51it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.85it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "9701b66cfa2746acbeb05eac5a3138a9",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 7 (246 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  9.57it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.98it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 955.86it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 111.05it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.86it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.63it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 987.13it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 122.40it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.29it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "e8a1ce3cd71542eaae64292eebb63e04",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 7 (246 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.01it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.89it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 560.59it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 91.77it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.44it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.69it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 988.99it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 108.23it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.36it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "464d3404e5614e558cbd12de2a3b1311",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 7 (246 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 11.11it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  6.29it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 322.66it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 115.67it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.82it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.48it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 919.00it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 105.16it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.60it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2eebbb8b2bb447fe91ad56c139424e86",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 7 (246 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  9.33it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.19it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 379.06it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 118.72it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.51it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.67it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 882.45it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 117.65it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.88it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "3ae51e4461be47b8af7adacc0d0683b9",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 7 (246 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 10.54it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.38it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 430.49it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 89.32it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.12it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.27it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 901.61it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 89.59it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.05it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2269f8534ef34b07ae717a14dda99f53",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 8 (399 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.015569).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 17.26it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.28it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 934.35it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 95.01it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 10.92it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.58it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 740.91it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 96.27it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.01it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "a4aeace2603e49de9f71af8eade1f74c",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 8 (399 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.003799).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 15.98it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.34it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 967.99it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 97.41it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.11it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.43it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 928.35it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 100.77it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.07it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "e108447b42174323b79bde8316236004",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 8 (399 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.005769).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 11.95it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.41it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 918.39it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 94.95it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.35it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.18it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 930.41it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 84.59it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.72it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "3b83a3aa53994c678c5769e5265df5f2",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 8 (399 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.002809).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 15.95it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.52it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 966.21it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 101.18it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.51it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.27it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 921.83it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 53.83it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.49it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0631dd9e5c6a40df8cbda3a36924d0df",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 8 (399 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.004891).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.12it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.43it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 336.92it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 145.61it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 15.34it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.41it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 950.66it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 138.54it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 16.43it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "069fe72af7754e6b8de6ab8f5a50fbfc",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 9 (183 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.52it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.48it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 557.68it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 175.44it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 15.73it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.45it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 964.21it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 176.18it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 16.63it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "3393b6876f0a4205afcf5d398825cfe2",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 9 (183 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.90it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.60it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 994.15it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 177.87it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 16.03it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.61it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 996.27it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 156.59it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 16.42it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "219c752da0494a7da4bcbe9fc5d1e6af",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 9 (183 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.09it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.71it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 984.58it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 149.97it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 15.14it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.49it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 990.16it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 146.95it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.88it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "235ba8d07f22422383e79e2c180e89f0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 9 (183 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.66it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.64it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 993.20it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 140.75it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.96it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.70it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 804.43it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 160.64it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 15.50it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "8ed336a6f76e4e46a1a6f5c702c58c00",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 9 (183 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.29it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.04it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 340.72it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 111.37it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.49it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.09it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 855.28it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 123.99it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.81it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "59c8243d465142d5aeae6905417adbc5",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 10 (289 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.008135).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 17.32it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.26it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 953.47it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 115.86it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.67it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.30it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 844.60it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 113.87it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.56it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d853c5b203284f5bad54d9eb8afdbd09",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 10 (289 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.004675).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 17.39it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.24it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 750.73it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 116.40it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.04it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.98it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 857.03it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 116.03it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.70it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "c721b212dc2143bc901c1544fd7e5527",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 10 (289 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.006083).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 17.33it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.93it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 801.82it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 104.76it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.63it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.07it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 848.88it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 106.44it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.74it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "4a191fd79d4547abb0f8fa002dffb877",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 10 (289 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.003889).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.96it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.10it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 805.67it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 132.98it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.51it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.38it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 794.53it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 117.75it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.69it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "95e99da1938a4361a99834ed73191557",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 10 (289 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.007405).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 13.73it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.64it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 436.09it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 130.52it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.03it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.58it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 960.89it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 125.09it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.61it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "65718d9ad31d44c6a62d030bb2459937",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 11 (277 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.000156).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 13.75it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.73it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 1017.54it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 122.64it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 16.18it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.61it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 905.12it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 128.09it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.28it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d67dd05b12d045e8b4e60f182bffcd2d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 11 (277 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.002388).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 16.69it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.72it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 988.99it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 123.43it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 15.69it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.87it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 952.17it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 141.87it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.41it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "71de5b57b4a049e7a327f0892f6248d7",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 11 (277 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.000832).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 19.44it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.74it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 947.01it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 132.19it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.06it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.68it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 985.04it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 143.99it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.88it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "91a07c47be774c86ba80e09e7b6ad255",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 11 (277 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.001847).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.14it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.67it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 786.04it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 102.62it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.49it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.79it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 985.27it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 130.12it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 15.94it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "69c625c716e54753ad013f7e937bdb63",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 11 (277 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.009914).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.84it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.88it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 318.35it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 123.69it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.06it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.10it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 835.35it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 125.99it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.28it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "61af7722a6034cc9810b507db96b9816",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 12 (223 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tEstimated heritability is out of bounds.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  9.41it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.93it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 829.08it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 153.71it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.94it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.11it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 500.45it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 134.42it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.21it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "ef13feddb2f242f482ca6c5489e4edc4",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 12 (223 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tEstimated heritability is out of bounds.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 14.00it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.16it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 597.05it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 110.93it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.40it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.78it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 780.63it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 128.40it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.59it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "67b41e4bcc84411a9b862f1fe3f29e27",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 12 (223 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tEstimated heritability is out of bounds.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  7.76it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.46it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 840.88it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 132.22it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.91it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.07it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 999.12it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 152.11it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 12.44it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "fa24380fc903425696df85c2224d2a90",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 12 (223 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tEstimated heritability is out of bounds.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.64it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  7.61it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 489.93it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 108.73it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 13.65it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  5.18it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 936.44it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 126.02it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 14.26it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2cbc64943beb4697a6fbde3de3b5b20b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 12 (223 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tEstimated heritability is out of bounds.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  5.81it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 10.37it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 321.25it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 115.13it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 16.86it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.00it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 900.07it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 137.23it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 18.08it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "554dd52d903a4b63abf39abdd07a85c7",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 13 (229 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.002286).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.06it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 10.97it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 920.01it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 132.22it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 18.56it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.09it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 956.95it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 142.53it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 17.57it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "612f24a2523b4e0ca4a8ea4dde3b9e74",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 13 (229 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tThe MSE is negative (-0.000598).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 11.16it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.13it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 1014.10it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 125.02it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 17.65it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.18it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 976.56it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 127.42it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 18.55it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "0ce02c74ceb94036b86fb927fca24c13",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 13 (229 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.001227).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 10.91it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.76it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 791.08it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 131.70it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 17.92it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.10it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 958.70it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 149.32it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 18.31it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "e7fff1dcfc784c5b8e8617d78f74b267",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 13 (229 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.004951).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 20.80it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 10.86it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 1000.79it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 125.33it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 16.94it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 10.18it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 805.98it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 128.75it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 18.16it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "8610396cadb84fc081f77d5282c47959",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 13 (229 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:545: RuntimeWarning: invalid value encountered in log\n",
      "  elbo -= np.log(2 * np.pi * self.sigma_epsilon)\n",
      "\tThe MSE is negative (-0.000351).\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 15.53it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.60it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 373.86it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 189.46it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.90it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.43it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 863.38it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 160.41it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.26it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "fc0d378d0d494c6b8ef9bdfd00076f2f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 14 (152 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.13it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.86it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 898.52it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 202.64it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.23it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.80it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 775.72it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 197.01it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 16.95it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "bf0c6df2eb5b4de29b6bafd63da620ff",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 14 (152 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.45it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.99it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 902.58it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 197.95it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.19it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.69it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 908.25it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 177.72it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 19.46it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "e2d05a41b6a344a78dd9fa5943d36091",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 14 (152 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 13.94it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.88it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 877.10it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 211.15it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 21.23it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.04it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 862.85it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 178.03it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 19.39it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "14b6994be56b4eddb7d4580dfa7728f6",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 14 (152 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 13.48it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.98it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 900.07it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 155.34it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.12it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.87it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 841.55it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 176.18it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 19.59it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f7d35480245e4e7e9f6ca99a25bde1f4",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 14 (152 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 13.02it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.60it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 365.10it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 184.24it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 21.84it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.64it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 848.36it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 186.99it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 21.35it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "20f27efad66c4e729d4a706f0772fd40",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 15 (149 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 11.71it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 10.09it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 1085.76it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 114.71it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 17.45it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.79it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 617.63it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 139.49it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 18.78it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "486f273220c043718e3beb5c8edb268a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 15 (149 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 18.49it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.35it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 833.36it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 164.63it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 17.19it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.20it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 586.86it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 141.47it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 19.44it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "bae4c5f8711a4baa92f608d741d0751a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 15 (149 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 18.34it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 10.53it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 544.93it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 145.28it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.51it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.55it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 689.17it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 166.55it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 19.02it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "cffd3c6657f845179adc348ab1371eba",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 15 (149 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tMaximum iterations reached without convergence.\n",
      "You may need to run the model for more iterations.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 24.10it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.65it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 67.46it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 57.70it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 16.22it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.14it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 728.81it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 192.68it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 21.42it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "979bafc4c4464ab59ea7cee910eeef6c",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 15 (149 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 17.47it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  9.38it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 200.96it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 134.23it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.59it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.80it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 1030.04it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 167.30it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 21.96it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f9ae66d3b4534eddad710fa8738ca5d0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 16 (128 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 22.92it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00,  8.82it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 1021.51it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 161.27it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 21.51it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.10it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 933.73it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 162.96it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.71it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "3ffc4ad03d574d298e5ed3855a7742dc",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 16 (128 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 16.07it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.54it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 1032.06it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 154.84it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.03it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.06it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 997.69it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 169.26it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 21.63it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "debfa801409f4d4e81b5bcbaf1ffbf2a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 16 (128 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 13.42it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.50it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 1014.10it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 188.79it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.36it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.88it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 937.07it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 160.70it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 22.59it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "26ee19524a6942c7a29ec384447ce223",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 16 (128 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.40it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.87it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 979.29it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 158.25it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 15.44it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.00it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 1064.27it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 189.93it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 21.68it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "5b859ada976e4e839f2a9bfa02976204",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 16 (128 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 11.98it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 14.62it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 462.95it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 185.60it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.81it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 16.62it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 801.97it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 184.09it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.54it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f491c7ff10b04cf08bfc867e9b3d37ef",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 17 (96 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:563: RuntimeWarning: divide by zero encountered in log\n",
      "  elbo -= np.multiply(null_gamma, np.log(null_gamma) - np.log(null_pi)).sum(axis=sum_axis)\n",
      "\tObjective (ELBO) is undefined.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 35.31it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 14.85it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 822.25it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 139.84it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.55it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 16.07it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 868.03it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 196.77it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.61it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "a8c66c540fbb4a9c8836ddfe458f8e4b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 17 (96 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:563: RuntimeWarning: divide by zero encountered in log\n",
      "  elbo -= np.multiply(null_gamma, np.log(null_gamma) - np.log(null_pi)).sum(axis=sum_axis)\n",
      "\tObjective (ELBO) is undefined.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 35.75it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 14.58it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 869.47it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 187.34it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.54it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 16.30it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 938.11it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 201.81it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.89it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "606f203960214cb8b2469f3f7d8023a6",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 17 (96 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:563: RuntimeWarning: divide by zero encountered in log\n",
      "  elbo -= np.multiply(null_gamma, np.log(null_gamma) - np.log(null_pi)).sum(axis=sum_axis)\n",
      "\tObjective (ELBO) is undefined.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 36.30it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.67it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 877.10it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 174.09it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.85it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 16.43it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 832.70it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 165.89it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 22.94it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "e7d6ff53b1744d3f84ff81a94283b228",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 17 (96 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:563: RuntimeWarning: divide by zero encountered in log\n",
      "  elbo -= np.multiply(null_gamma, np.log(null_gamma) - np.log(null_pi)).sum(axis=sum_axis)\n",
      "\tObjective (ELBO) is undefined.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 39.02it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 14.81it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 1006.07it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 168.38it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.59it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 16.38it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 920.01it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 222.18it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 22.99it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "887653e56b90445a8ca522e08626d580",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 17 (96 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Library/Frameworks/Python.framework/Versions/3.12/lib/python3.12/site-packages/viprs/model/VIPRS.py:563: RuntimeWarning: divide by zero encountered in log\n",
      "  elbo -= np.multiply(null_gamma, np.log(null_gamma) - np.log(null_pi)).sum(axis=sum_axis)\n",
      "\tObjective (ELBO) is undefined.\n",
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 32.72it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.48it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 368.05it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 169.63it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 19.74it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.77it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 925.89it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 165.67it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.13it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "23c418bd423541b6b61bb80a3308fb97",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 18 (135 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 29.40it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.23it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 923.25it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 152.34it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 21.29it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.50it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 859.84it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 177.27it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.30it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2d5e2c69e608401aa89c920ed0c84a99",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 18 (135 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 15.98it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.42it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 882.83it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 201.62it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.08it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.55it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 899.29it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 158.75it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.40it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "e28685f47fc4470c8b63c2130064eb71",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 18 (135 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 15.79it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 12.49it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 938.95it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 206.32it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 21.53it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.82it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 943.39it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 204.87it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.18it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "491e6dbb00cc480ea6bddebef175f3e6",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 18 (135 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  7.71it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.41it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 878.76it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 207.60it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 19.52it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.30it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 958.70it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 176.49it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 20.19it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f48c6aa1c3bf4bdb9515a7cc71292a94",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 18 (135 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 19.09it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 22.62it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 519.23it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 155.88it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 28.29it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 21.84it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 1033.08it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 164.29it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 29.60it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "65c23d4ffe034cbd939e584775785d5a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 19 (82 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 18.18it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 18.62it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 978.61it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 161.52it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 28.98it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 24.53it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 931.03it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 169.34it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 27.92it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d7247c87f42f4712ae1ba985516b3ec1",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 19 (82 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 51.21it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 21.90it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 867.31it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 78.98it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 25.41it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 25.53it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 925.69it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 162.82it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 31.32it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "d72893f9a3244bbd889e6ec05fc2d630",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 19 (82 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 43.28it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 20.10it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 204.88it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 143.24it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 30.18it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 24.19it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 910.82it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 184.98it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 28.45it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2eb2fe2df38e49a7985048533b2cf786",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 19 (82 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 51.74it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 18.39it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 486.13it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 180.08it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 29.32it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 24.18it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 838.19it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 166.75it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 31.08it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "1727ee13763f45e29ac1621c924b9fd5",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 19 (82 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 42.56it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.05it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 317.75it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 177.15it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 24.23it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 15.24it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 841.89it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 218.73it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 24.25it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2a74d8fbe15f4778862642cf95fdaabb",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 20 (119 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 14.22it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 14.28it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 873.45it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 199.07it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 24.14it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 15.68it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 692.59it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 186.32it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 25.55it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "a4f2f60e34084b2ebcdf496ebc8e6e6c",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 20 (119 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 12.36it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 13.97it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 819.04it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 172.83it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.49it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 15.35it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 914.19it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 174.70it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 24.56it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "a6f8e155b51b4209a439a82b41efac4a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 20 (119 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 13.91it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 11.13it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 996.98it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 149.99it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 24.58it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 15.51it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 862.14it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 220.36it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 24.15it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "e621a1ae4b12418c8f79265c3e2f910d",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 20 (119 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 10.44it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 14.94it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 979.29it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 211.68it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 23.98it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 15.56it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 806.44it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 191.14it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 22.64it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "737932966e0b4b1793b299ed8b70ca58",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 20 (119 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 30.66it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 21.86it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 490.22it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 312.77it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 32.31it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 23.60it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 921.83it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 261.46it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 34.72it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "96f33cb2e0f044aca029a15d8fcc585a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 21 (44 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 20.98it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 21.36it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 824.19it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 303.01it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 28.00it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 22.08it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 819.04it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 299.76it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 34.67it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "831967dffb6f4ec9bda30853f43f3132",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 21 (44 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 52.02it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 18.52it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 909.04it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 232.41it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 33.73it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 23.27it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 875.64it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 209.34it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 34.50it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "96a988e280f04ddcbe7e5400720b7d20",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 21 (44 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 46.67it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 20.05it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 494.49it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 170.53it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 34.19it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 24.12it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 820.32it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 260.08it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 32.07it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "720dd1a863204c60af2806d41857d044",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 21 (44 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 47.81it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 21.67it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 734.68it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 189.75it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 29.44it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 23.87it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 768.05it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 313.78it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 33.58it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "195e47f79ac94ecf9f4744e99806cbf4",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 21 (44 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 50.53it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 23.42it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 475.06it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 271.07it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 35.99it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 19.63it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 903.36it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 240.66it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 36.16it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "98d979a903d3431f8915c3fbcb8bec66",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 22 (58 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 45.18it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 15.80it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 519.68it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 166.34it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 35.15it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 24.63it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 965.10it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 266.37it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 39.58it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "24b4f8ec274b43c39a98e16ea6cc9c32",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 22 (58 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00,  9.75it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 16.34it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 724.53it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 306.38it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 35.31it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 24.74it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 696.27it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 284.59it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 33.94it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "80772ffc50a949949f7f6bae3fb439ba",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 22 (58 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 15.48it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 19.19it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 854.76it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 240.04it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 37.22it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 24.47it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 856.85it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 262.46it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 38.49it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "67c1c56f397542a3be997783db382040",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 22 (58 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 41.69it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 21.80it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 366.60it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 283.84it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 36.98it/s]\n",
      "Reading genotype metadata: 100%|██████████| 1/1 [00:00<00:00, 25.67it/s]\n",
      "Reading LD metadata: 100%|██████████| 1/1 [00:00<00:00, 768.05it/s]\n",
      "Reading summary statistics: 100%|██████████| 1/1 [00:00<00:00, 290.95it/s]\n",
      "Harmonizing data: 100%|██████████| 1/1 [00:00<00:00, 38.24it/s]\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "133e7e221d2e4e219c055436ad6afeb0",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "Chromosome 22 (58 variants):   0%|          | 0/1000 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Generating polygenic scores: 100%|██████████| 1/1 [00:00<00:00, 46.81it/s]\n"
     ]
    }
   ],
   "source": [
    "np.random.seed(42)\n",
    "df_all = pd.DataFrame({'ID': sample_ids})\n",
    "for i in range(1, 23):\n",
    "    df = get_cv_prs(i)\n",
    "    df_all = df_all.merge(df, on=\"ID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_all.to_csv(\"Japan_prs_test2.csv\", index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
